### PR TITLE
[Improvement][server] Fix some sonar analyzed bugs

### PR DESCRIPTION
**Pull request by CalvinKirs**
fix assert error(assertions always pass) .
use AtomicReference<Object> to replace unsafe volatile Object
lazy initialization holder class Singleton...(unsafe volatile Object)
#3321

### CODE REVIEW

**Comment by CalvinKirs** [comment]
TBD.

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ResourceMapperTest.java [hunk 1]
```diff
@@ -76,8 +76,10 @@ private Resource insertOne(){
         resource.setDirectory(false);
         resource.setType(ResourceType.FILE);
         resource.setUserId(111);
-        resourceMapper.insert(resource);
-        return resource;
+        if( resourceMapper.insert(resource)>0 ){
+            return resource;
+        }
+        throw new RuntimeException("insert data error");
```
Comment (gabrywu) [inline dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ResourceMapperTest.java:82]: If the RuntimeException, how the invokers to handle it ?

Comment (yangyichao-mango) [reply dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ResourceMapperTest.java:82]: I also think assert.fail() is better.

**Comment by CalvinKirs** [comment]
Hi, thank you very much for your review. I noticed that many of our codes did not judge the success status of the relevant execution results when performing DDL-related operations. The changes here will be larger. Do I need to create a new PR to complete it?

**Comment by CalvinKirs** [comment]
Thanks for your review, I will finish it later




On 08/6/2020 09:32，Yichao Yang<notifications@github.com> wrote：

@yangyichao-mango commented on this pull request.

In dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ResourceMapperTest.java:

> @@ -76,8 +76,10 @@ private Resource insertOne(){
         resource.setDirectory(false);
         resource.setType(ResourceType.FILE);
         resource.setUserId(111);
-        resourceMapper.insert(resource);
-        return resource;
+        if( resourceMapper.insert(resource)>0 ){
+            return resource;
+        }
+        throw new RuntimeException("insert data error");


I also think assert.fail() is better.

File: dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ResourceMapperTest.java (hunk 1 above)
Comment (CalvinKirs) [reply dolphinscheduler-dao/src/test/java/org/apache/dolphinscheduler/dao/mapper/ResourceMapperTest.java:82]: Thanks for your review, I will finish it later

**Comment by CalvinKirs** [comment]
To Be Done

**Comment by CalvinKirs** [comment]
done

**Comment by zhuangchong** [comment]
+1

**Comment by yangyichao-mango** [comment]
Please change title according the specifications and format your code with checkstyle and code-style configurations.

**Comment by CalvinKirs** [comment]
> Please change title according the specifications and format your code with checkstyle and code-style configurations.
> 
> Checkstyle.xml configuration lesson: https://www.cnblogs.com/wanshi1989/p/5478050.html
> code style xml configuration lesson: https://blog.csdn.net/qq_34579060/article/details/80100040

It is recommended to implement in the new file, the old file will cause confusion in the submission record.

**Comment by yangyichao-mango** [comment]
> > Please change title according the specifications and format your code with checkstyle and code-style configurations.
> > Checkstyle.xml configuration lesson: https://www.cnblogs.com/wanshi1989/p/5478050.html
> > code style xml configuration lesson: https://blog.csdn.net/qq_34579060/article/details/80100040
> 
> It is recommended to implement in the new file, the old file will cause confusion in the submission record.

I think it will be better to implement in new pr, but at least format the changed code.

**Comment by CalvinKirs** [comment]
> > > Please change title according the specifications and format your code with checkstyle and code-style configurations.
> > > Checkstyle.xml configuration lesson: https://www.cnblogs.com/wanshi1989/p/5478050.html
> > > code style xml configuration lesson: https://blog.csdn.net/qq_34579060/article/details/80100040
> > 
> > 
> > It is recommended to implement in the new file, the old file will cause confusion in the submission record.
> 
> I think it will be better to implement in new pr, but at least format the changed code.
thanks,I will finish it later

**Comment by CalvinKirs** [comment]
> > > Please change title according the specifications and format your code with checkstyle and code-style configurations.
> > > Checkstyle.xml configuration lesson: https://www.cnblogs.com/wanshi1989/p/5478050.html
> > > code style xml configuration lesson: https://blog.csdn.net/qq_34579060/article/details/80100040
> > 
> > 
> > It is recommended to implement in the new file, the old file will cause confusion in the submission record.
> 
> I think it will be better to implement in new pr, but at least format the changed code.

done.

**Comment by yangyichao-mango** [comment]
Please change title to `[Improvement][server] Fix some sonar analyzed bugs`. Thx a lot~

**Comment by CalvinKirs** [comment]
> Please change title to `[Improvement][server] Fix some sonar analyzed bugs`. Thx a lot~

done.

**Review by gabrywu — APPROVED** [review]
+1

**Review by dailidong — APPROVED** [review]
+1
good job

**Pull request merged**